### DISCUSSION TITLE
Smart transition not working(?)

### DISCUSSION THREAD

**Discussion by Parakovsky** [savonet/liquidsoap#3515] [User Support]
Reading this amazing book third time I notice the authors indicate there are feauture called 'smart' transition, which if one modify crossfade (smart=true) will manage to get all the marvels of it. It works but not in all tracks, in fact only 1/3 of music got crossfades, they all the same format - lossless, but variable bitrate (compression?) maybe decoding encoding process somehow messed things up? Maybe smart option isn't suppose to work at all? Here is the relevant error from the logs. 

2023/11/09 18:49:59 [crossfade:3] old >= new + margin, new >= medium and old <= high
2023/11/09 18:49:59 [crossfade:3] Old source is significantly louder than new one.
2023/11/09 18:49:59 [crossfade:3] Transition: crossed, fade-in.

here it is obviously works, and works great. 

2023/11/09 18:47:19 [crossfade:3] No transition: using default.

here when it fail, why?

everybody, have a great day whenever you are and thank you for this amazing software.

**Comment by toots** [comment]
Hi @Parakovsky ! When you say it fails, what do you mean exactly? What were you expecting and what are you observing?

**Comment by Parakovsky** [comment]
> Hi @Parakovsky ! When you say it fails, what do you mean exactly? What were you expecting and what are you observing?

Hello, I meant it fails in sense there is no crossfade happening between songs, which is I suppose a normal behaviour for some rules of the program but I don't know which one to look at.

**Reply by toots** [reply]
Yeah that's expected for some transitions. If you want to crossfade all the time you can either use the simple transitions or else write your own custom smart transition! See the doc here: https://www.liquidsoap.info/doc-2.2.2/crossfade.html#example

**Reply by Parakovsky** [reply]
Ok I figured out what it was, it was replaygain! If you cut it off the transitions works much better, must be the math behind it is too precise for system to figure out so it thinks everything is even even when it is not. I didn't like how it sounds with replaygain anyway, are there any differences in algos to it? Smart transition without it sounds much better anyway.
